src/fastcs/transport/epics/p4p/types.py

```diff
@@ -0,0 +1,191 @@
+import math
+import time
+from dataclasses import asdict
+
+import numpy as np
+from numpy.typing import DTypeLike
+from p4p.nt import NTEnum, NTNDArray, NTScalar, NTTable
+
+from fastcs.attributes import Attribute
+from fastcs.datatypes import Bool, Enum, Float, Int, String, T, Table, Waveform
+
+P4P_ALLOWED_DATATYPES = (Int, Float, String, Bool, Enum, Waveform, Table)
+
+
+_P4P_EXTRA = [("description", ("u", None, [("defval", "s")]))]
+_P4P_BOOL = NTScalar("?", extra=_P4P_EXTRA)
+_P4P_STRING = NTScalar("s", extra=_P4P_EXTRA)
+
+
+_P4P_EXTRA_NUMERICAL = [
+    ("units", ("u", None, [("defval", "s")])),
+    ("min", ("u", None, [("defval", "d")])),
+    ("max", ("u", None, [("defval", "d")])),
+    ("min_alarm", ("u", None, [("defval", "d")])),
+    ("max_alarm", ("u", None, [("defval", "d")])),
```

Comment:
Please can you put these in their standard place?
Probably `limitLow` and `limitHigh` are the right place?

https://github.com/epics-base/p4p/blob/9040d4bb56d58b674de1475dbb62444a056f6729/src/p4p/nt/scalar.py#L115

I think we probably just ignore the alarm limits for now, as that's a muddy area:
https://docs.epics-controls.org/en/latest/pv-access/Normative-Types-Specification.html#alarmlimit-t